sdk/src/ed25519_instruction.rs

```diff
@@ -94,6 +98,9 @@ pub fn verify(
     let expected_data_size = num_signatures
         .saturating_mul(SIGNATURE_OFFSETS_SERIALIZED_SIZE)
         .saturating_add(SIGNATURE_OFFSETS_START);
+    if data[1] != 0 && feature_set.is_active(&check_ed25519_instruction_padding::id()) {
```

Comment:
I see it is checked above but still iffy to use direct indexing, maybe use `get`